### PR TITLE
Add basic authentication function

### PR DESCRIPTION
**Pull request by jyx11011**

### CODE REVIEW

File: src/main/webapp/js/util.js [hunk 1]
```diff
@@ -14,3 +14,24 @@ function addLoadEvent(func) {
     window.onload = func;
   }
 }
+
+function configureHeaderLoginStatus() {
+	fetch("/api/authentication")
```
Comment (limli) [inline src/main/webapp/js/util.js]: don't use tabs. Run eslint on this file.

File: src/main/webapp/WEB-INF/components/header.html [hunk 2]
```diff
@@ -27,7 +27,12 @@
         <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="/idk">idk</a></li>
       </ul>
       <ul class="navbar-nav">
-        <li class="nav-item mx-0 mx-lg-1"><a class="nav-link py-3 px-0 px-lg-3 rounded" href="/login">Login</a></li>
+        <li id="login-status" class="nav-item mx-0 mx-lg-1">
+          <a id="login-status-link" class="nav-link py-3 px-0 px-lg-3 rounded" href="#"></a>
+        </li>
+        <li id="login-status" class="nav-item mx-0 mx-lg-1">
+          <a id="header-username" class="nav-link py-3 px-0 px-lg-3 rounded" href="#"></a>
+        </li>
```
Comment (anuragshirolkar) [inline src/main/webapp/WEB-INF/components/header.html]: Why are there two nodes with the same id (`login-status`)? Maybe it should be a class?

**Review by anuragshirolkar — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/components/header.html (hunk 2 above)
Comment (SuyashLakhotia) [reply src/main/webapp/WEB-INF/components/header.html]: +1

Comment (jyx11011) [reply src/main/webapp/WEB-INF/components/header.html]: Sorry, I deleted the ids, since they are not used anywhere.

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**